Carbon.Redis/Abstractions/ICarbonCacheExtensions.cs

```diff
@@ -83,39 +127,164 @@ public static async Task HashSetAsObject<T>(this ICarbonCache instance, string k
+        /// <param name="instance"></param>
+        /// <param name="key">Redis Key</param>
+        /// <param name="setMethodIfNotExists">Execute if this key does not exist</param>
+        /// <returns>Your entire object</returns>
+        public static async Task<T> HashGetAsObject<T>(this ICarbonCache instance, string key, Func<Task<T>> setMethodIfNotExists) where T : class
+        {
+            var value = await instance.HashGetAsObject<T>(key);
+
+            if(value == default)
+            {
+                T result = await setMethodIfNotExists();
+                await instance.HashSetAsObject<T>(key, result);
+                return result;
+            }
+
+            return value;
+        }
+
+        /// <summary>
+        /// Set an expiration time for your redis key (TTL) Time-To-Live
+        /// </summary>
+        /// <param name="instance"></param>
+        /// <param name="key">Redis Key</param>
+        /// <param name="expiryTimeSpan">Absolute Time Span starting as of now</param>
+        /// <returns>Successful or Failed response directly from Redis</returns>
+        /// <exception cref="NotSupportedException">Expiry Time cannot be set to 0</exception>
         public static async Task<bool> SetTTL(this ICarbonCache instance, string key, TimeSpan expiryTimeSpan)
         {
             if (expiryTimeSpan == TimeSpan.Zero || expiryTimeSpan == default(TimeSpan))
             {
                 throw new NotSupportedException("Expiry Time cannot be set to 0");
             }
-            return await instance.GetDatabase().KeyExpireAsync(key, expiryTimeSpan);
+            return await instance.GetDatabase().KeyExpireAsync(key.ToRedisInstanceKey(instance), expiryTimeSpan);
+        }
+
+        /// <summary>
+        /// Set an expiration time for your redis key (TTL) Time-To-Live
+        /// </summary>
+        /// <param name="instance"></param>
+        /// <param name="key">Redis Key</param>
+        /// <param name="expiryTimeSpan">Absolute Time Span starting as of now</param>
+        /// <returns>Successful or Failed response directly from Redis</returns>
+        /// <exception cref="NotSupportedException">Expiry Time cannot be set to 0</exception>
+        public static async Task<bool> SetTTL(this Task<(ICarbonCache, string)> instanceWithKey, TimeSpan expiryTimeSpan)
+        {
+            var awaitedInstanceWithKey = await instanceWithKey;
+            if (expiryTimeSpan == TimeSpan.Zero || expiryTimeSpan == default(TimeSpan))
+            {
+                throw new NotSupportedException("Expiry Time cannot be set to 0");
+            }
+            return await awaitedInstanceWithKey.Item1.GetDatabase().KeyExpireAsync(awaitedInstanceWithKey.Item1.GetInstanceName() + awaitedInstanceWithKey.Item2, expiryTimeSpan);
+        }
+
+        /// <summary>
+        /// Simple set a hash in redis with a single object where your content will be a single hash field in the hash key
```

Comment:
Why are you starting summaries with "simple" :D It had me confused for a second, because I was reading "single" instead of "simple"

Reply:
Did you mean to write "simply" maybe?

Reply:
Actually it is really "Simple Set", or in other words "Basic Set Operation" or sth like that